lib/fog/azurerm/storage.rb

```diff
@@ -32,20 +46,36 @@ def initialize(_options = {})
           end
         end
       end
+
       # This class provides the actual implemention for service calls.
       class Real
         def initialize(options)
           begin
             require 'azure_mgmt_storage'
             require 'azure/storage'
+            @debug = ENV["DEBUG"] || options[:debug]
+            require "azure/core/http/debug_filter" if @debug
           rescue LoadError => e
             retry if require('rubygems')
             raise e.message
           end
 
           credentials = Fog::Credentials::AzureRM.get_credentials(options[:tenant_id], options[:client_id], options[:client_secret])
-          @storage_mgmt_client = ::Azure::ARM::Storage::StorageManagementClient.new(credentials)
-          @storage_mgmt_client.subscription_id = options[:subscription_id]
+          unless credentials.nil?
+            @storage_mgmt_client = ::Azure::ARM::Storage::StorageManagementClient.new(credentials)
+            @storage_mgmt_client.subscription_id = options[:subscription_id]
+          end
+
+          if Fog::Credentials::AzureRM.new_account_credential? options
```

Comment:
Thanks for reviewing.

Let me explain more about the reason I am adding code here. In the Azure Storage REST API, there are two categories of APIs:

- **Resource management APIs**: Targeting on creating/deleting/updating/fetching storage accounts
- **Storage data APIs**: Targeting on the storage blob/table/queue/file operations

With this understanding, I assume there are two scenarios when users create a `Fog::Storage::AzureRM` instance. One is for managing the storage account and the other is for managing the blob/table/queue/file resources. 

For the Azure storage data operations, there could be always be intensive calls. So I am considering optimizing the calls:
- Reducing REST calls. I may not get the blob client as the code in `delete_disk` does, which makes an additional call to list the accounts and searches for the account key.
- Reuse the client instance. As the intensive calls against the storage account, I'd like to reuse the client instance. That's why I create a instance variable `@blob_client` in the `Fog::Storage::AzureRM`. I will also add `@table_client`, `@queue_client` and `@file_client` in the similar way, if you agree with this approach.

On the other hand, it is a common practice that the user is not a subscription admin and he/she doesn't have a client ID and subscription ID. What the user has are the storage account name and key or a SAS token. When the user creates a `Fog::Storage::AzureRM` instance, getting the data client from the object he/she gets looks an intuitive way.

Does it make sense to you?